### PR TITLE
LPS-130364 Migrate selection modals in commerce-product-asset-categories-web

### PR DESCRIPTION
**Pull request by kresimir-coko**
Hey @liferay-commerce @riccardo-alberti 

This is a part of an [ongoing epic](https://issues.liferay.com/browse/LPS-129246) to replace legacy AUI-based modals with Clay/React-based `openSelectionModal`. In this PR, we are replacing modals in `commerce-product-asset-categories-web`.

JIRA: https://issues.liferay.com/browse/LPS-130364

Previously reviewed at https://github.com/liferay-frontend/liferay-portal/pull/1021

# 📜  Notes
- There are parts of this interaction that should be improved, but since they're not a part of the epic I'll leave those to you. For example: 
    - Modal not closing after choosing the category (I assume that is the intended behaviour but I might be wrong)
    - [The visuals](https://prnt.sc/12snrj7) leave something to be desired - consistency between the headings, and labels once something is selected
    - Both modals (Select Category modal should either be `single` or `multiple` selection, not a mix that it is currently)
- I've made a change in UI behaviour regarding the `Select Category Display Page` modal. Previously it was single selection with footer buttons to confirm or cancel - since it's a single selection modal, it no longer has the footer

# 🧪 Testing
1. Product Menu > Commerce > Channels
2. Create or edit a Channel
3. Go to Category Display Pages
4. Add a new Display Layout

### CODE REVIEW

**Comment by kresimir-coko** [comment]
ci:test:sf

**Comment by kresimir-coko** [comment]
ci:test:relevant

**Comment by kresimir-coko** [comment]
ci:test:relevant

**Comment by kresimir-coko** [comment]
ci:close

**Comment by kresimir-coko** [comment]
ci:reopen

**Comment by kresimir-coko** [comment]
ci:test:relevant

**Comment by jonmak08** [comment]
ci:test:relevant

File: modules/apps/commerce/commerce-product-asset-categories-web/package.json [hunk 1]
```diff
@@ -0,0 +1,13 @@
+{
+	"dependencies": {
+		"frontend-js-web": "*"
+	},
+	"name": "@liferay/commerce-product-asset-categories-web",
+	"scripts": {
+		"build": "liferay-npm-scripts build",
+		"checkFormat": "liferay-npm-scripts check",
+		"format": "liferay-npm-scripts fix",
+		"test": "liferay-npm-scripts test"
```
Comment (markocikos) [inline modules/apps/commerce/commerce-product-asset-categories-web/package.json:10]: @kresimir-coko this line is causing the CI failure, it is trying to run frontend tests and we don't have any. We can just remove it.

**Review by markocikos — CHANGES_REQUESTED** [review]

File: modules/apps/commerce/commerce-product-asset-categories-web/package.json (hunk 1 above)
Comment (kresimir-coko) [reply modules/apps/commerce/commerce-product-asset-categories-web/package.json:10]: 👍  thank you!

**Comment by kresimir-coko** [comment]
I've rebased with `master`, fixed conflicts, and removed the test script from `package.json` as mentioned in https://github.com/liferay-commerce/liferay-portal/pull/1155#discussion_r633424947

**Comment by kresimir-coko** [comment]
ci:test:relevant

**Comment by kresimir-coko** [comment]
@FabioDiegoMastrorilli can you please get this one reviewed?

File: modules/apps/commerce/commerce-product-asset-categories-web/src/main/resources/META-INF/resources/js/EditAssetCategoryCPDisplayLayout.js [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+
+						if (categoryNodes.length) {
+							for (let i = 0; i < categoryNodes.length; i++) {
+								categoriesContainer.removeChild(
+									categoryNodes.item(i)
+								);
+							}
+						}
+
+						delete selectedItems.key;
```
Comment (FabioDiegoMastrorilli) [inline modules/apps/commerce/commerce-product-asset-categories-web/src/main/resources/META-INF/resources/js/EditAssetCategoryCPDisplayLayout.js:182]: Hi @kresimir-coko, this was already there but is wrong - it's basically removing the same `key` property each single loop. Could you please 🙏  update it in `delete selectedItems[key];`

**Review by FabioDiegoMastrorilli — CHANGES_REQUESTED** [review]
Done! Great job (I just asked for little fix on something old)

File: modules/apps/commerce/commerce-product-asset-categories-web/src/main/resources/META-INF/resources/js/EditAssetCategoryCPDisplayLayout.js (hunk 2 above)
Comment (kresimir-coko) [reply modules/apps/commerce/commerce-product-asset-categories-web/src/main/resources/META-INF/resources/js/EditAssetCategoryCPDisplayLayout.js]: @FabioDiegoMastrorilli Fixed in https://github.com/liferay-commerce/liferay-portal/pull/1155/commits/ad5ef278751287411c1175832fdeafb5f1794d09

I've also rebased with `master`

**Comment by kresimir-coko** [comment]
@FabioDiegoMastrorilli bump

**Comment by FabioDiegoMastrorilli** [comment]
ci:forward

**Comment by markocikos** [comment]
> FabioDiegoMastrorilli commented 20 hours ago
> ci:forward

CI still pending? This can't be right :thinking: I'll try restarting it.

**Comment by markocikos** [comment]
ci:stop

**Comment by markocikos** [comment]
ci:close

**Comment by markocikos** [comment]
ci:reopen

**Comment by kresimir-coko** [comment]
`junit.framework.AssertionFailedError: Version must match the project version (4.0.4) modules/apps/commerce/commerce-product-asset-categories-web/package.json`

Looks to me like the code is too stale, gonna rebase with master and push

**Comment by kresimir-coko** [comment]
ci:test:sf

**Comment by kresimir-coko** [comment]
ci:test:relevant

**Comment by kresimir-coko** [comment]
ci:test:sf

**Comment by kresimir-coko** [comment]
ci:test:relevant

**Comment by kresimir-coko** [comment]
I have no idea why the CI is complaining here, do we have any QA to look over it, pretty please?

**Comment by kresimir-coko** [comment]
ci:test:relevant

**Comment by kresimir-coko** [comment]
This failure unique to this pull looks unrelated to the PR, it is talking about Documents & Media, specifically the Display Style picker which works fine for me.
image

**Comment by markocikos** [comment]
ci:test:relevant

**Comment by markocikos** [comment]
ci:forward

**Pull request closed without merging**